### ISSUE TITLE
mpHealth-1.0 throws IllegalStateException during server startup

### ISSUE THREAD

**Issue by navidsh** [OpenLiberty/open-liberty#9574]
On a Liberty `19.0.0.9` server with `mpHealth-1.0` enabled, when the application is loading, if `/health` is invoked `IllegalStateException` can happen.

I have a WAR application packaged in an EAR. The EAR includes multiple apps that can take a bit of time to load. 

- `server.xml` includes the following:
```xml
	<application type="ear" id="SocialSphere_mpcfg" name="SocialSphere_mpcfg" location="${server.config.dir}/apps/SocialSphereOpenAPI31_mpcfg.ear">
		<application-bnd>
	 	   <security-role name="admin_role">
                     <user name="persona1"/>
                     <user name="persona2"/>
		     <user name="persona3"/>
		   </security-role>
		</application-bnd>	
	</application>
```

- Exception:
```
Stack Dump = java.lang.IllegalStateException: WEB#SocialSphereOpenAPI31_mpcfg#AllergyForecastMPOpenAPI.war
	at com.ibm.ws.container.service.metadata.internal.MetaDataServiceImpl.getMetaData(MetaDataServiceImpl.java:268)
	at com.ibm.ws.classloading.internal.ClassLoadingServiceImpl.getClassLoaderIdentifier(ClassLoadingServiceImpl.java:707)
	at com.ibm.ws.microprofile.health.services.impl.AppModuleContextService.getContextClassLoaderIdentifier(AppModuleContextService.java:145)
	at com.ibm.ws.microprofile.health.services.impl.AppModuleContextService.createThreadContext(AppModuleContextService.java:168)
	at com.ibm.ws.microprofile.health.services.impl.AppModuleContextService.createContextualProxy(AppModuleContextService.java:194)
	at com.ibm.ws.microprofile.health.services.impl.HealthExecutorImpl.runHealthChecks(HealthExecutorImpl.java:70)
	at com.ibm.ws.microprofile.health.internal.HealthCheckServiceImpl.performHealthCheck(HealthCheckServiceImpl.java:108)
	at com.ibm.ws.microprofile.health.internal.servlet.HealthCheckServlet.service(HealthCheckServlet.java:53)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:182)
...
```

- `HEALTH=all` traces: [mphealth-trace.zip](https://github.com/OpenLiberty/open-liberty/files/3790856/mphealth-trace.zip)

**Comment by pgunapal** [comment]
Thanks Navid for the detailed information!

Looks like the issue occurs when applications are slow to start up and when multiple concurrent requests are made to the `/health` endpoint, before the application starts up, for mpHealth-1.0.

This issue does not occur for mpHealth-2.0, as it was addressed and fixed in PR https://github.com/OpenLiberty/open-liberty/pull/7894/files

We are discussing a potential fix for this issue for mpHealth-1.0, will keep the issue updated.

**Comment by pgunapal** [comment]
@navidsh would it be possible it to always use mpHealth-2.0, instead of 1.0, since the fix is included in mpHealth-2.0, and it has better APIs and endpoints, to serve the user's needs better?

Is there a specific test case that requires the use of mpHealth-1.0?

**Comment by navidsh** [comment]
We'll switch from `mpHealth-1.0` to `2.0`. So fixing this issue is not strategic for us anymore. I'll close this issue. Thanks
